pkg/kubelet/dockertools/fake_docker_client.go

```diff
@@ -74,25 +75,58 @@ type FakeDockerClient struct {
 const fakeDockerVersion = "1.8.1"
 
 func NewFakeDockerClient() *FakeDockerClient {
-	return NewFakeDockerClientWithVersion(fakeDockerVersion, minimumDockerAPIVersion)
+	return &FakeDockerClient{
+		VersionInfo:  dockertypes.Version{Version: fakeDockerVersion, APIVersion: minimumDockerAPIVersion},
+		Errors:       make(map[string]error),
+		ContainerMap: make(map[string]*dockertypes.ContainerJSON),
+		Clock:        clock.RealClock{},
+		// default this to an empty result, so that we never have a nil non-error response from InspectImage
+		Image: &dockertypes.ImageInspect{},
+		// default this to true, so that we trace calls, image pulls and container lifecycle
+		EnableTrace: true,
+	}
 }
 
-func NewFakeDockerClientWithClock(c clock.Clock) *FakeDockerClient {
-	return newClientWithVersionAndClock(fakeDockerVersion, minimumDockerAPIVersion, c)
+func (f *FakeDockerClient) WithClock(c clock.Clock) *FakeDockerClient {
+	f.Clock = c
+	return f
 }
 
-func NewFakeDockerClientWithVersion(version, apiVersion string) *FakeDockerClient {
```

Comment:
There are some other places using this function. You also need to update them.
For example https://github.com/kubernetes/kubernetes/blob/8fd414537b5143ab039cb910590237cabf4af783/pkg/kubelet/dockertools/images_test.go#L27.